akka-docs/rst/scala/code/docs/http/scaladsl/server/directives/TimeoutDirectivesExamplesSpec.scala

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (C) 2009-2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package docs.http.scaladsl.server.directives
+
+import akka.http.scaladsl.model.{HttpResponse, StatusCodes}
+import akka.http.scaladsl.server.RoutingSpec
+
+import scala.concurrent.duration._
+import scala.concurrent.{Future, Promise}
+
+class TimeoutDirectivesExamplesSpec extends RoutingSpec {
+
+  "Request Timeout" should {
+    "be configurable in routing layer" in {
+      pending // compile only spec since requires actuall Http server to be run
+
+      //#withRequestTimeout-plain
+      val route =
+        path("timeout") {
+          withRequestTimeout(3.seconds) {
+            val response: Future[String] = slowFuture() // very slow
+            complete(response)
+          }
+        }
+      //#
+    }
+    "without timeout" in {
+      pending // compile only spec since requires actuall Http server to be run
```

Comment:
in some specs we have there is a `compileTimeOnly(block: => Any)` that is pretty nice